modules/gdscript/gdscript_parser.cpp

```diff
@@ -837,6 +837,7 @@ GDScriptParser::Node *GDScriptParser::_parse_expression(Node *p_parent, bool p_s
 						if (scr.is_valid() && scr->is_valid()) {
 							ConstantNode *constant = alloc_node<ConstantNode>();
 							constant->value = scr;
+							constant->datatype = _type_from_variant(constant->value);
```

Comment:
There are other code blocks nearby where a `ConstantNode` is allocated and the `datatype` member is not set. Should it always be set as here?

Reply:
most of the the constant's value(`Variant::Type`) is used for type checking, but It's essential to initialize the datatype as well.
@akien-mga